src/Documentation/Advanced-Concepts/Dependency-Injection.md

```diff
@@ -109,4 +109,52 @@ public class InjectedService : IInjectedService
         return Task.FromResult(DateTime.UtcNow.Ticks);
     }
 }
-```
\ No newline at end of file
+```
+
+# Test Framework Integration
+
+DI truly shines when coupled with a testing framework to verify the correctness of the code that build. You can read about the components for making testing in Orleans by following our [tutorials](../../Tutorials/Unit-Testing-Grains.md).
+
+You will need to do two things setup DI with tests. First you will need to implement mocks of your services. Here is an example of doing this.
+
+
+``` csharp
+public class MockServices
+{
+    public IServiceProvider ConfigureServices(IServiceCollection services)
+    {
+        var mockInjectedService = new Mock<IInjectedService>();
```

Comment:
Is it ever mentioned which mocking framework is being used? It might be worthwhile to point out that "this example uses Moq" or something like that.